navigator/backend/src/main/scala/com/digitalasset/navigator/json/ApiCodecCompressed.scala

```diff
@@ -34,35 +35,45 @@ object ApiCodecCompressed {
     case v: Model.ApiEnum => apiEnumToJsValue(v)
     case v: Model.ApiList => apiListToJsValue(v)
     case Model.ApiText(v) => JsString(v)
-    case Model.ApiInt64(v) => JsString(v.toString)
-    case Model.ApiDecimal(v) => JsString(v)
+    case Model.ApiInt64(v) => JsString((v: Long).toString)
+    case Model.ApiDecimal(v) => JsString(v.decimalToString)
     case Model.ApiBool(v) => JsBoolean(v)
-    case Model.ApiContractId(v) => JsString(v.toString)
+    case Model.ApiContractId(v) => JsString(v)
     case t: Model.ApiTimestamp => JsString(t.toIso8601)
     case d: Model.ApiDate => JsString(d.toIso8601)
-    case Model.ApiParty(v) => JsString(v.toString)
-    case Model.ApiUnit() => JsObject.empty
+    case Model.ApiParty(v) => JsString(v)
+    case Model.ApiUnit => JsObject.empty
     // Note: Optional needs to be boxed, otherwise the following values are indistinguishable:
     // None, Some(None), Some(Some(None)), ...
     case Model.ApiOptional(None) => JsObject(fieldNone -> JsObject.empty)
     case Model.ApiOptional(Some(v)) => JsObject(fieldSome -> apiValueToJsValue(v))
     case v: Model.ApiMap =>
       apiMapToJsValue(v)
+    case _: Model.ApiImpossible => serializationError("impossible! tuples are not serializable")
   }
 
-  def apiListToJsValue(value: Model.ApiList): JsValue =
-    JsArray(value.elements.map(apiValueToJsValue).toVector)
-
-  def apiVariantToJsValue(value: Model.ApiVariant): JsValue =
-    JsObject(Map(value.constructor -> apiValueToJsValue(value.value)))
-
-  def apiEnumToJsValue(value: Model.ApiEnum): JsValue =
-    JsString(value.constructor)
-
-  def apiRecordToJsValue(value: Model.ApiRecord): JsValue =
-    JsObject(value.fields.map(f => f.label -> apiValueToJsValue(f.value)).toMap)
+  private[this] def apiListToJsValue(value: Model.ApiList): JsValue =
+    JsArray(value.values.map(apiValueToJsValue).toImmArray.toSeq: _*)
```

Comment:
Could be a candidate to use `CanBuildFrom`.
```suggestion
    JsArray(value.values.map(apiValueToJsValue)(FrontStack.`FrontStack canBuildFrom`): _*)
```